packages/open-next/src/build/createAssets.ts

```diff
@@ -0,0 +1,261 @@
+      rsc: files.rsc ? fs.readFileSync(files.rsc, "utf8") : undefined,
+      body: files.body
+        ? fs
+            .readFileSync(files.body)
+            .toString(
+              isBinaryContentType(cacheFileMeta.headers["content-type"])
+                ? "base64"
+                : "utf8",
+            )
+        : undefined,
+    };
+    fs.writeFileSync(cacheFilePath, JSON.stringify(cacheFileContent));
+  });
+
+  if (!config.dangerous?.disableTagCache) {
+    // Generate dynamodb data
+    // We need to traverse the cache to find every .meta file
+    const metaFiles: {
+      tag: { S: string };
+      path: { S: string };
+      revalidatedAt: { N: string };
+    }[] = [];
+
+    // Compute dynamodb cache data
+    // Traverse files inside cache to find all meta files and cache tags associated with them
+    buildHelper.traverseFiles(
+      outputPath,
+      (file) => file.endsWith(".meta"),
+      (filePath) => {
+        const fileContent = fs.readFileSync(filePath, "utf8");
+        const fileData = JSON.parse(fileContent);
+        if (fileData.headers?.["x-next-cache-tags"]) {
+          fileData.headers["x-next-cache-tags"]
+            .split(",")
+            .forEach((tag: string) => {
+              // TODO: We should split the tag using getDerivedTags from next.js or maybe use an in house implementation
+              metaFiles.push({
+                tag: { S: path.posix.join(buildId, tag.trim()) },
+                path: {
+                  S: path.posix.join(
+                    buildId,
+                    path.relative(outputPath, filePath).replace(".meta", ""),
+                  ),
+                },
+                // We don't care about the revalidation time here, we just need to make sure it's there
+                revalidatedAt: { N: "1" },
+              });
+            });
+        }
+      },
+    );
+
+    // Copy fetch-cache to cache folder
+    const fetchCachePath = path.join(
+      appBuildOutputPath,
+      ".next/cache/fetch-cache",
+    );
+    if (fs.existsSync(fetchCachePath)) {
+      const fetchOutputPath = path.join(outputDir, "cache", "__fetch", buildId);
+      fs.mkdirSync(fetchOutputPath, { recursive: true });
+      fs.cpSync(fetchCachePath, fetchOutputPath, { recursive: true });
+
+      buildHelper.traverseFiles(
+        fetchCachePath,
+        () => true,
+        (filepath) => {
+          const fileContent = fs.readFileSync(filepath, "utf8");
+          const fileData = JSON.parse(fileContent);
+          fileData?.tags?.forEach((tag: string) => {
+            metaFiles.push({
+              tag: { S: path.posix.join(buildId, tag) },
+              path: {
+                S: path.posix.join(
+                  buildId,
+                  path.relative(fetchCachePath, filepath),
+                ),
+              },
+              revalidatedAt: { N: "1" },
+            });
+          });
+        },
+      );
+    }
+
+    if (metaFiles.length > 0) {
```

Comment:
We could probably split everything below as well. We could create a `compileTagCacheProvider` function. 
It would get rid of the eslint warning as well

Reply:
I've added a TODO.
I prefer this PR to be mostly code move and we can iterate in follow-up PRs.